### PR TITLE
Add GatewayBot

### PR DESCRIPTION
**Pull request by jonas747**

### CODE REVIEW

**Comment by jonas747** [comment]
so the test fails because it's a bot only endpoint, there is some other bot only and user only endpoints as well though, so probably gonna need 2 sessions for testing. should i remove the test for now?

File: restapi.go [hunk 1]
```diff
@@ -1685,6 +1685,28 @@ func (s *Session) Gateway() (gateway string, err error) {
 	return
 }
 
+// Gateway returns the websocket Gateway address and the reccomended number of shards
```
Comment (acdenisSK) [inline restapi.go]: s/reccomended/recommended

Comment (jonas747) [inline restapi.go]: No Hablo Ingles

**Comment by iopred** [comment]
@bwmarrin could you create a new bot account so we can add this test?

**Comment by bwmarrin** [comment]
I can do that :)

**Comment by bwmarrin** [comment]
I just added a Bot account under the dgtest account.  I added the token to travis with the env `DGB_TOKEN`

**Comment by jonas747** [comment]
It's still complaining it's a bot only endpoint

**Comment by bwmarrin** [comment]
I think the code needs to change so that it uses the dgbot session that your other PR added.

**Comment by bwmarrin** [comment]
Hmn, well that token is definitely a Bot account.  I added the "Bot " prefix to the travis settings and now it's got a new error.  HTTP 401 Unauthorized, {"code": 0, "message": "401: Unauthorized"}

**Comment by jonas747** [comment]
Have you connected to the gateway once with that token?

**Comment by bwmarrin** [comment]
I've ran the discordgo ping pong example to test it.. You can of course try it locally, it shows the token in the Travis details link for the failed build

**Comment by jonas747** [comment]
Ran fine using that token on my machine 
image

**Pull request merged**